shm/src/Ship.cpp

```diff
@@ -0,0 +1,27 @@
+#include "../inc/Ship.hpp"
+
+Ship::Ship(size_t id, std::string name, size_t speed, size_t maxCrew, size_t capacity)
+    : id_(id)
+    , name_(name)
+    , speed_(speed)
+    , maxCrew_(maxCrew)
+    , capacity_(capacity)
+{}
+
+Ship::Ship(size_t id, size_t speed, size_t maxCrew) 
+    : Ship(id, "Ship", speed, maxCrew, 100)
+{}
+
+Ship& Ship::operator+=(const size_t value){
+    crew_ += value;
+    return *this;
+}
+
+Ship& Ship::operator-=(const size_t value){
+    crew_ -= value;
```

Comment:
What if crew_ becomes negative?

size_t in an unsigned datatype thus cannot be negative (crew_ count neither)

Reply:
> What if crew_ becomes negative?
> 
> size_t in an unsigned datatype thus cannot be negative (crew_ count neither)

Good point